app/src/main/java/com/example/sofascoreapp/viewmodel/TournamentsViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package com.example.sofascoreapp.viewmodel
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.sofascoreapp.data.model.Tournament
+import com.example.sofascoreapp.data.networking.Network
+import kotlinx.coroutines.launch
+import retrofit2.Response
+
+class TournamentsViewModel : ViewModel() {
+
+    private val _footballTournaments = MutableLiveData<Response<ArrayList<Tournament>>>()
```

Comment:
slicna stvar kao i za evente, ove livedate nemaju potrebu znati o kojem tocno sportu je rijec. One primaju listu turnira, i to je sve sto one trebaju znati. Jedino je bitno napraviti tocan request za sport. Tako da cijeli ovaj viewmodel mozes krace napisati ovako, bez dupliciranja koda:

```Kotlin
private val _tournaments = MutableLiveData<Response<ArrayList<Tournament>>>()

    fun setTournaments(results: Response<ArrayList<Tournament>>) {
        _tournaments.value = results
    }

    fun getTournaments(): MutableLiveData<Response<ArrayList<Tournament>>> {
        return _tournaments
    }

    fun getTournaments(sport: String) {
        viewModelScope.launch {
            setTournaments(Network().getService().getTournamentsForSport(sport))
        }
    }
```